src/tools/clippy/clippy_lints/src/methods/unnecessary_to_owned.rs

```diff
@@ -435,7 +435,7 @@ fn can_change_type<'a>(cx: &LateContext<'a>, mut expr: &'a Expr<'a>, mut ty: Ty<
                         let output_ty = fn_sig.output();
                         if output_ty.contains(*param_ty) {
                             if let Ok(new_ty)  = cx.tcx.try_subst_and_normalize_erasing_regions(
-                                new_subst, cx.param_env, output_ty) {
+                                new_subst, cx.param_env, EarlyBinder(output_ty)) {
```

Comment:
So, seems like the `subst_identity().skip_binder()` call to create this isn't correct...

Reply:
Hmm, yeah I agree that this looks weird now. Would it be better as `skip_binder().skip_binder()`? Or are you envisioning a larger change here?

Reply:
I have to look a bit more at this. But likely the easiest thing is to either just leave it as-is with a comment, or map/skip_binder above.

Reply:
Ah okay thanks! I changed it to `skip_binder()` above in the most recent commit, but happy to try to improve it further.

Yeah, I also thought about mapping it, it seems like we always skip the inner `Binder` there but want to keep the `EarlyBinder` around, so could do something like

```rust
let fn_sig = cx.tcx.fn_sig(def_id).map_bound(|t| t.skip_binder());
```

(and then replace the calls to  `.inputs()` with `.skip_binder().inputs()`)

but I wasn't sure if this was more or less readable than how it is now

Reply:
I think this PR doesn't necessarily need to fix this, at minimum we should probably open a clippy issue about it and add a FIXME before this PR lands